### PR TITLE
fix: fixing typescript vinyl types

### LINKED ISSUE #122: Broken types
Hi, types seems to be still invalid. If you use gulp-replace with typescript in version 1.1.2 it throws error:

```
Only ECMAScript imports may use 'import type'.

1 /// <reference types="node" />
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 import type File = require("vinyl");
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 ```
 
 Seems that for type imports you cannot use `require`, so code should look like:
 
 ```
 /// <reference types="node" />
import File = require("vinyl");
```

OR:

```
/// <reference types="node" />
import type * as File from "vinyl";
```

It looks like d.ts files not need `type` imports, so first solution could be better.

### PR DESCRIPTION
**Pull request by wszydlak**
fix #122

### CODE REVIEW

**Comment by wszydlak** [comment]
I've found better solution - dynamic vinyl import type.

**Comment by lazd** [comment]
TypeScript is hard. @manuth @y805939188 @jon2180 please verify, thanks!

File: index.d.ts [hunk 1]
```diff
@@ -1,5 +1,4 @@
 /// <reference types="node" />
-import type File = require("vinyl");
```
Comment (manuth) [inline index.d.ts:2]: Instead of moving the `require('vinyl')` down to the `ReplacerContext`, just remove the `type`-keyword here.

**Review by manuth — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution!

Sorry for not providing backwards compatibility - honestly I didn't know that the `type`-keyword isn't compatible with older TS-versions.

File: index.d.ts (hunk 1 above)
Comment (wszydlak) [reply index.d.ts:2]: Done ;)

**Review by manuth — APPROVED** [review]
Awesome, thanks a lot!
Looks good to me.

**Comment by wszydlak** [comment]
@lazd can we have this soon merged and released?

**Comment by lazd** [comment]
Released as 1.1.3!

**Pull request merged**